### ISSUE TITLE
Add change_ring method to polygons and surfaces

### ISSUE THREAD

**Issue by slel** [flatsurf/sage-flatsurf#96]
Polynomials, vectors and matrices have a change_ring method.

It would be nice to have one for polygons and surfaces too,
returning a new polygon or surface over the new base ring
(with an option to change the polygon or surface in place).

**Comment by wphooper** [comment]
You can do this through the `copy` method:
``` python
from flatsurf import *
s = translation_surfaces.regular_octagon()
ss = s.copy(new_field=AA)
```

Perhaps `new_field` should be `new_ring` or `base_ring`.

**Comment by saraedum** [comment]
I am working on making `SimilaritySurface` a proper parent with unique representation. I think adding an explicit `change_ring` then makes sense. In particular, since I want to deprecate `copy` as part of deprecating support for mutable `SimilaritySurface`.

**Comment by saraedum** [comment]
Fixed in #220
